rust/windows-client/src-tauri/build.rs

```diff
@@ -1,3 +1,18 @@
 fn main() {
-    tauri_build::build()
+    let win = tauri_build::WindowsAttributes::new().app_manifest(WINDOWS_MANIFEST);
+    let attr = tauri_build::Attributes::new().windows_attributes(win);
+    tauri_build::try_build(attr).expect("build.rs");
 }
+
+// If we ask for admin privilege in the manifest, we can't run in the CLI,
+// which makes debugging hard.
```

Comment:
why is this the case?

Reply:
I suspect it's because the app manifest is only loaded for the GUI bundle and not for raw binaries.

Headless / CLI mode operation isn't the highest priority in the short-term, but when we get to it I think it's fair to require users to right-click the exe to run as admin (or the equivalent powershell version of that for automation).

We may even decide it's best to have a separate build script or crate to generate a slimmed down single binary without all the Tauri cruft.

But again, not high priority at the moment -- all demand for the Windows app is for end-user workstations.

Reply:
My mistake, it works from Powershell. It's only Cygwin that gets a "permission denied"

By "GUI" and "CLI" here I mean that Windows makes a weird distinction - CLI programs can open GUI windows, but they have a console window open by default, which you wouldn't want for a production GUI. GUI programs don't have stdout and stderr by default and it's painful to set it up.

So to make development easier, I'm following the Tauri default of building it as a CLI (that happens to open windows) for debug and a GUI for release.

The distinction is some silly thing where GUI programs start in WinMain instead of main, probably dates back to the Windows 1.0 days. https://github.com/rust-lang/rfcs/blob/master/text/1665-windows-subsystem.md

Reply:
Cleaned up this comment in b6a5bb7d

Reply:
Makes sense